docs/examples/tensorflow/tensorflow-resnet50-various-readers.ipynb

```diff
@@ -0,0 +1,286 @@
+    "                                            crop = (227, 227),\n",
+    "                                            image_type = types.RGB,\n",
+    "                                            mean = [128., 128., 128.],\n",
+    "                                            std = [1., 1., 1.])\n",
+    "        self.uniform = ops.Uniform(range = (0.0, 1.0))\n",
+    "\n",
+    "    def base_define_graph(self, inputs, labels):\n",
+    "        images = self.decode(inputs)\n",
+    "        images = self.resize(images)\n",
+    "        output = self.cmn(images, crop_pos_x = self.uniform(),\n",
+    "                          crop_pos_y = self.uniform())\n",
+    "        return (output, labels.gpu())"
+   ]
+  },
+  {
+   "cell_type": "markdown",
+   "metadata": {},
+   "source": [
+    "- MXNetReaderPipeline"
+   ]
+  },
+  {
+   "cell_type": "code",
+   "execution_count": null,
+   "metadata": {},
+   "outputs": [],
+   "source": [
+    "from nvidia.dali.pipeline import Pipeline\n",
+    "import nvidia.dali.ops as ops\n",
+    "import nvidia.dali.types as types\n",
+    "\n",
+    "class MXNetReaderPipeline(CommonPipeline):\n",
+    "    def __init__(self, batch_size, num_threads, device_id, num_gpus):\n",
+    "        super(MXNetReaderPipeline, self).__init__(batch_size, num_threads, device_id)\n",
+    "        self.input = ops.MXNetReader(path = [db_folder+\"train.rec\"], index_path=[db_folder+\"train.idx\"],\n",
+    "                                     random_shuffle = True, shard_id = device_id, num_shards = num_gpus)\n",
+    "\n",
+    "    def define_graph(self):\n",
+    "        images, labels = self.input(name=\"Reader\")\n",
+    "        return self.base_define_graph(images, labels)"
+   ]
+  },
+  {
+   "cell_type": "markdown",
+   "metadata": {},
+   "source": [
+    "- FileReadPipeline"
+   ]
+  },
+  {
+   "cell_type": "code",
+   "execution_count": null,
+   "metadata": {},
+   "outputs": [],
+   "source": [
+    "class FileReadPipeline(CommonPipeline):\n",
+    "        def __init__(self, batch_size, num_threads, device_id, num_gpus):\n",
+    "            super(FileReadPipeline, self).__init__(batch_size, num_threads, device_id)\n",
+    "            self.input = ops.FileReader(file_root = image_dir)\n",
+    "\n",
+    "        def define_graph(self):\n",
+    "            images, labels = self.input()\n",
+    "            return self.base_define_graph(images, labels)"
+   ]
+  },
+  {
+   "cell_type": "markdown",
+   "metadata": {},
+   "source": [
+    "- TFRecordPipeline"
+   ]
+  },
+  {
+   "cell_type": "code",
+   "execution_count": null,
+   "metadata": {},
+   "outputs": [],
+   "source": [
+    "import nvidia.dali.tfrecord as tfrec\n",
+    "\n",
+    "class TFRecordPipeline(CommonPipeline):\n",
+    "    def __init__(self, batch_size, num_threads, device_id, num_gpus):\n",
+    "        super(TFRecordPipeline, self).__init__(batch_size, num_threads, 0)\n",
```

Comment:
Creating multiple TFRecordPipeline assigned to different device_id was failing for me, so I always assign it to 0. I will look more into that issue.

Reply:
I must have done something wrong. It works now.